vcpkg-configuration.json

```diff
@@ -0,0 +1,6 @@
+{
+  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/vcpkg-configuration.schema.json",
+  "overlay-ports": [
+    "./Scripts/Ports"
+   ]
```

Comment:
Since we are doing this, we can remove this line https://github.com/colincornaby/Plasma/blob/bare-apple-client/cmake/VcpkgToolchain.cmake#L48

Reply:
I'll take a look.

I've noticed caching doesn't seem to work with our custom packages. I think Python specifically is recompiling every time. Which is slowing down builds. Does that seem right to you, and if caching isn't working is there a way to get it working again?

Reply:
I'm not sure if overlay ports have any known issues or limitations regarding binary caching. I wonder if it would make more sense to pull our overlay ports out into our own [registry](https://learn.microsoft.com/en-us/vcpkg/maintainers/registries), which would make our ports look just like they are part of vcpkg itself. I'm going to ask in the vcpkg discord channel if overlay ports use binary caching.

Reply:
As an aside, I also noticed that VcpkgToolchain.cmake needs to be updated to select the plasma triplets by default on macOS.

Reply:
I thought those triplets were automatically applied. They may not be necessary any more with all the other vcpkg changes. I'll look into it and maybe just yank them if they're not needed.

Reply:
I removed the triplets for macOS. I added them when I was trying to get Python to build statically. After reviewing the vcpkg docs I noted that static is the default in the built in macOS triplet, so I don't think we need the macOS triplets anymore. We can revisit if the triplets ever do become needed.

However - the one setting that needs to be maintained is the VCPKG macOS deployment target. I moved this into the main toolchain as it's the same across all macOS versions, and the presence of the define should not affect other platforms. If that setting really does belong in a triplet, let me know.